lib/Doctrine/Migrations/Tools/Console/Command/MigrateCommand.php

```diff
@@ -106,88 +109,102 @@ protected function configure() : void
 
 Use the --all-or-nothing option to wrap the entire migration in a transaction.
 EOT
-        );
+            );
 
         parent::configure();
     }
 
     public function execute(InputInterface $input, OutputInterface $output) : ?int
     {
         $this->outputHeader($output);
-
-        $version          = (string) $input->getArgument('version');
+        $allowNoMigration = $input->getOption('allow-no-migration');
+        $versionAlias     = $input->getArgument('version');
         $path             = $input->getOption('write-sql');
-        $allowNoMigration = (bool) $input->getOption('allow-no-migration');
-        $timeAllQueries   = (bool) $input->getOption('query-time');
-        $dryRun           = (bool) $input->getOption('dry-run');
-        $allOrNothing     = $this->getAllOrNothing($input->getOption('all-or-nothing'));
-
-        $this->configuration->setIsDryRun($dryRun);
 
-        $version = $this->getVersionNameFromAlias(
-            $version,
-            $output
-        );
+        try {
+            $version = $this->getDependencyFactory()->getVersionAliasResolver()->resolveVersionAlias($versionAlias);
+        } catch (UnknownMigrationVersion|NoMigrationsFoundWithCriteria $e) {
+            $this->getVersionNameFromAlias($versionAlias, $output);
 
-        if ($version === '') {
             return 1;
         }
 
-        if ($this->checkExecutedUnavailableMigrations($input, $output) === 1) {
+        $planCalculator                = $this->getDependencyFactory()->getMigrationPlanCalculator();
+        $executedUnavailableMigrations = $planCalculator->getExecutedUnavailableMigrations();
+
+        if ($this->checkExecutedUnavailableMigrations($executedUnavailableMigrations, $input, $output) === false) {
+            return 3;
+        }
+
+        $migratorConfigurationFactory = $this->getDependencyFactory()->getConsoleInputMigratorConfigurationFactory();
+        $migratorConfiguration        = $migratorConfigurationFactory->getMigratorConfiguration($input);
+
+        $plan = $planCalculator->getPlanUntilVersion($version);
+
+        /**
+         * If there are no migrations to execute throw an exception.
+         */
```

Comment:
It does not look like an exception is thrown

Reply:
True, just status 1 is returned